lib/services/schema-object-factory.ts

```diff
@@ -310,6 +311,14 @@ export class SchemaObjectFactory {
     if (typeDefinitionRequiredFields.length > 0) {
       typeDefinition['required'] = typeDefinitionRequiredFields;
     }
+
+    if (schemas[schemaName] && !isEqual(schemas[schemaName], typeDefinition)) {
+      throw new Error(
```

Comment:
I wonder if, for the time being, we could just log an error instead of throwing an exception. Can throw an exception in the next major release then

Reply:
done